File: trainer-app-backend/src/main/java/uns/ac/rs/trainerappbackend/dto/TrainingDto.java
package uns.ac.rs.trainerappbackend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class TrainingDto {

    private Long id;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private String status;
    private String duration;
    private LocalDateTime createdAt;
    private LocalDateTime canceledAt;
    private UserTrainingDto user;
    private TrainerTrainingDto trainer;

}

File: trainer-app-fronted/src/app/modules/training/types/TrainingDto.ts
import { TrainerTrainingDto } from "./TrainerTrainingDto";
import { UserTrainingDto } from "./UserTrainingDto";

export interface TrainingDto {
  id: number;
  startTime: string;  
  endTime: string;
  status: string;
  duration: string;
  createdAt: string;
  canceledAt?: string | null;
  user?: UserTrainingDto | null;
  trainer: TrainerTrainingDto;
}